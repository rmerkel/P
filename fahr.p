{ print Fahrenheit-Celsius table	}
{	first version; integers only	}

const
	LOWER =   0;	{	lower table limit	}
	UPPER = 300;	{	upper table limit	}
	STEP  =  20;	{	table step size		}

var
	fahr, celsius : integer;

begin
	fahr = LOWER;
	while fahr <= UPPER do begin
		celsius = 5 * (fahr-32) / 9;
		fahr = fahr + STEP;
	end;
 end.
