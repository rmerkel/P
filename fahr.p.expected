# fahr.p, 2: { print Fahrenheit-Celsius table	}
# fahr.p, 3: {	first version; integers only	}
# fahr.p, 4: 
# fahr.p, 5: const
    0: call 0, 2
    1: halt
# fahr.p, 6: 	LOWER =   0;	{	lower table limit	}
# fahr.p, 7: 	UPPER = 300;	{	upper table limit	}
# fahr.p, 8: 	STEP  =  20;	{	table step size		}
# fahr.p, 9: 
# fahr.p, 10: var
# fahr.p, 11: 	fahr, celsius : integer;
# fahr.p, 12: 
# fahr.p, 13: begin
    2: enter 2
# fahr.p, 14: 	fahr = LOWER;
    3: pushConst 0
    4: pushVar 0, 4
    5: assign
# fahr.p, 15: 	while fahr <= UPPER do begin
    6: pushVar 0, 4
    7: eval
    8: pushConst 300
    9: lte
   10: jneq 28
# fahr.p, 16: 		celsius = 5 * (fahr-32) / 9;
   11: pushConst 5
   12: pushVar 0, 4
   13: eval
   14: pushConst 32
   15: subi
   16: muli
   17: pushConst 9
   18: divi
   19: pushVar 0, 5
   20: assign
# fahr.p, 17: 		fahr = fahr + STEP;
   21: pushVar 0, 4
   22: eval
   23: pushConst 20
   24: addi
   25: pushVar 0, 4
   26: assign
# fahr.p, 18: 	end;
   27: jump 6
# fahr.p, 19:  end.
   28: ret

        8:          0
        9: -       17
        8:         20
        9: -        6
        8:         40
        9:          4
        8:         60
        9:         15
        8:         80
        9:         26
        8:        100
        9:         37
        8:        120
        9:         48
        8:        140
        9:         60
        8:        160
        9:         71
        8:        180
        9:         82
        8:        200
        9:         93
        8:        220
        9:        104
        8:        240
        9:        115
        8:        260
        9:        126
        8:        280
        9:        137
        8:        300
        9:        148
        8:        320
